src/lib/cache/gdfs_cache.hpp

```diff
@@ -1,19 +1,23 @@
 #pragma once
 
 #include <list>
+#include <shared_mutex>
 #include <unordered_map>
 #include <utility>
 
-#include "abstract_cache_impl.hpp"
+#include "abstract_cache.hpp"
 #include "boost/heap/fibonacci_heap.hpp"
 #include "utils/assert.hpp"
 
 namespace opossum {
 
-// Generic cache implementation using the GDFS policy.
-// Note: This implementation is not thread-safe.
+/*
+ * Generic cache implementation using the GDFS policy.
```

Comment:
Add a comment that previous implementations exist, but have been removed in this PR (link it)?

Reply:
Should this rather be here or in `AbstractCache`?

Reply:
Doesn't really matter, I guess.